File: Strategy3/payment_context.py
class PaymentContext:
    """
    Contexto de pago que utiliza una estrategia de pago determinada.
    """

    def __init__(self, payment_strategy):
        """
        Constructor que establece la estrategia de pago.

        Args:
            payment_strategy (PaymentStrategy): Estrategia de pago.
        """
        self.payment_strategy = payment_strategy

    def perform_payment(self, amount):
        """
        Realiza el pago utilizando la estrategia seleccionada.

        Args:
            amount (float): Monto a pagar.
        """
        self.payment_strategy.pay(amount)

File: Strategy2/PaymentContext.java
package Strategy2;

// Contexto de pago que utiliza una estrategia de pago determinada
class PaymentContext {
    private PaymentStrategy paymentStrategy;

    // Establece la estrategia de pago
    public void setPaymentStrategy(PaymentStrategy paymentStrategy) {
        this.paymentStrategy = paymentStrategy;
    }

    // Realiza el pago utilizando la estrategia seleccionada
    public void performPayment(double amount) {
        paymentStrategy.pay(amount);
    }
}